src/overlays/actors/ovl_En_Elf/z_en_elf.c

```diff
@@ -21,99 +66,1505 @@ const ActorInit En_Elf_InitVars = {
+                            temp = 8;
+                            break;
+                        case 11:
+                            temp = this->unk_2A8;
+                            if (this->unk_2C0 > 0) {
+                                this->unk_2C0--;
+                            } else {
+                                temp = 0;
+                            }
+                            break;
+                        default:
+                            temp = 0;
+                            break;
+                    }
+                }
+            }
+        }
+
+        switch (temp) {
+            case 0:
+                if (!(player->stateFlags2 & 0x100000)) {
+                    temp = 7;
+                    if (this->unk_2C7 == 0) {
+                        Audio_PlayActorSound2(&this->actor, NA_SE_EV_NAVY_VANISH);
+                    }
+                }
+                break;
+            case 8:
+                if (player->stateFlags2 & 0x100000) {
+                    func_80A0299C(this, 0x32);
+                    this->unk_2C0 = 42;
+                    temp = 11;
+                    if (this->unk_2C7 == 0) {
+                        Audio_PlayActorSound2(&this->actor, NA_SE_EV_FAIRY_DASH);
+                    }
+                }
+                break;
+            case 7:
+                player->stateFlags2 &= ~0x100000;
+                break;
+            default:
+                player->stateFlags2 |= 0x100000;
+                break;
+        }
+    }
+
+    if (temp != this->unk_2A8) {
+        func_80A01C38(this, temp);
+
+        if (temp == 11) {
+            this->unk_2B8 = Math_Vec3f_DistXZ(&player->bodyPartsPos[8], &this->actor.posRot.pos);
+            this->unk_2AC = Math_Vec3f_Yaw(&this->actor.posRot.pos, &player->bodyPartsPos[8]);
+        }
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A0461C.s")
+#endif
+
+void EnElf_SpawnSparkles(EnElf* this, GlobalContext* globalCtx, s32 sparkleLife) {
+    static Vec3f sparkleVelocity = { 0.0f, -0.05f, 0.0f };
+    static Vec3f sparkleAccel = { 0.0f, -0.025f, 0.0f };
+    s32 pad;
+    Vec3f sparklePos;
+    Color_RGBA8 primColor;
+    Color_RGBA8 envColor;
+
+    sparklePos.x = Math_Rand_CenteredFloat(6.0f) + this->actor.posRot.pos.x;
+    sparklePos.y = (Math_Rand_ZeroOne() * 6.0f) + this->actor.posRot.pos.y;
+    sparklePos.z = Math_Rand_CenteredFloat(6.0f) + this->actor.posRot.pos.z;
+
+    primColor.r = this->innerColor.r;
+    primColor.g = this->innerColor.g;
+    primColor.b = this->innerColor.b;
+
+    envColor.r = this->outerColor.r;
+    envColor.g = this->outerColor.g;
+    envColor.b = this->outerColor.b;
+
+    EffectSsKiraKira_SpawnDispersed(globalCtx, &sparklePos, &sparkleVelocity, &sparkleAccel, &primColor, &envColor,
+                                    1000, sparkleLife);
+}
+
+void func_80A04D90(EnElf* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32* bgId;
```

Comment:
```suggestion
    s32 bgId;
```

Reply:
from mzxs work
```
f32 func_8003CA0C(GlobalContext* globalCtx, CollisionContext* colCtx, CollisionPoly** outPoly, s32* bgId, Actor* actor,
                  Vec3f* pos)
```

Reply:
you provide a pointer so that that function can write to it, and you can use it later (even tho it doesnt in this case)

Reply:
oh nvm i be dumb :)